notify/sns/sns.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2021 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sns
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"unicode/utf8"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+	commoncfg "github.com/prometheus/common/config"
+)
+
+// Notifier implements a Notifier for SNS notifications.
+type Notifier struct {
+	conf    *config.SNSConfig
+	tmpl    *template.Template
+	logger  log.Logger
+	client  *http.Client
+	retrier *notify.Retrier
```

Comment:
The `retrier` doesn't look used right now, but I think we should.

Reply:
We don't get a response code back in the event the publish is successful,[ for the final return ](https://github.com/treid314/alertmanager/blob/sns-reciever/notify/sns/sns.go#L159)should we be returning `false, nil` or `retrier(200, <response body from successful message>)`

Reply:
> We don't get a response code back in the event the publish is successful

Is the error a custom type exposing extra information, like if it's retryable or not?

Reply:
The error appears to be just `nil` returned from the sender in the success case and not wrapping anything.